.github/workflows/sdks.yml

```diff
@@ -189,76 +199,82 @@ jobs:
-          key: ${{ steps.version.outputs.key }}
+          key: ${{ steps.version.outputs.arch-key }}
       - name: Setup Swift ${{ matrix.version }} Android SDK
         id: sdk-setup
         run: |
           set -x
           cd sdk-config
-          tar xf ~/swift-${{ matrix.version }}-android-${{ matrix.arch }}*-sdk.tar.xz
-          pushd swift-*-sdk
-          SDK_PATH=${PWD}
-          popd
 
-          if ${{ startsWith(matrix.os, 'ubuntu') }}; then
-            ARCH_JSON=${SDK_PATH}/usr/swiftpm-android-${{ matrix.arch }}.json
-            cat ${ARCH_JSON}
-            NDK_PREBUILT=${ANDROID_NDK_LATEST_HOME}/toolchains/llvm/prebuilt/linux-x86_64
-          elif ${{ startsWith(matrix.os, 'macos') }}; then
+          ARCH_TARGET="${{ matrix.arch }}-unknown-linux-android${ANDROID_API_LEVEL}"
+          if ${{ matrix.arch == 'armv7' }}; then
+              ARCH_TARGET="armv7-unknown-linux-androideabi${ANDROID_API_LEVEL}"
+          fi
+
+          if ${{ steps.cache-bundle.outputs.cache-hit != 'true' }}; then
+            SDK_PATH=$(basename ~/swift-${{ matrix.version }}-android-${{ matrix.arch }}*-sdk.tar.xz .tar.xz)
+            tar xf ~/$SDK_PATH.tar.xz
+
             # fix the clang symlink
             ls -la ${SDK_PATH}/usr/lib/swift/clang
             rm -vf ${SDK_PATH}/usr/lib/swift/clang
             # Swift 5.10: usr/lib/clang/15.0.0 Trunk and Devel: usr/lib/clang/17
             ${TOOLCHAIN}/bin/clang --version
             ln -sfv ${TOOLCHAIN}/lib/clang/* ${SDK_PATH}/usr/lib/swift/clang
-            ls -d ${SDK_PATH}/usr/lib/swift/clang/include
-
-            # the toolchain destination JSON in the SDK is for ubuntu,
-            # so for macOS we need to manually build it
-            ARCH=${{ matrix.arch }}
-            ARCH_JSON="$PWD/android-${ARCH}.json"
-            ARCH_TARGET="${ARCH}-unknown-linux-android${ANDROID_API_LEVEL}"
-            ARCHID="${ARCH}-linux-android"
-            if [ "$ARCH" == "armv7" ]; then
-                ARCHID="arm-linux-androideabi"
-            fi
+            ls -l ${SDK_PATH}/usr/lib/swift/clang/lib/
 
-            NDK_PREBUILT=${ANDROID_NDK_LATEST_HOME}/toolchains/llvm/prebuilt/darwin-x86_64
+            ARCH_JSON="$PWD/android-${{ matrix.arch }}.json"
+
+            NDK_PREBUILT=${ANDROID_NDK_LATEST_HOME}/toolchains/llvm/prebuilt/${{ startsWith(matrix.os, 'macos') && 'darwin' || 'linux' }}-x86_64
+            if ${{ startsWith(matrix.os, 'macos') }}; then
+              cp -r ${NDK_PREBUILT}/lib/clang/*/lib/linux ${SDK_PATH}/usr/lib/swift/clang/lib
+            else
+              cp -r ${NDK_PREBUILT}/lib/clang/*/lib/linux/* ${SDK_PATH}/usr/lib/swift/clang/lib/linux
+            fi
 
             cat > ${ARCH_JSON} << EOF
             {
-                "version": 1,
-                "target": "${ARCH_TARGET}",
-                "toolchain-bin-dir": "${TOOLCHAIN}/bin",
-                "sdk": "${NDK_PREBUILT}/sysroot",
-                "extra-cc-flags": [ "-fPIC" ],
```

Comment:
@finagolfin I noticed I need to add this flag manually:

```
swift build --swift-sdk aarch64-unknown-linux-android24 --toolchain "/Library/Developer/Toolchains/swift-6.0.2-RELEASE.xctoolchain" -Xcc -fPIC
```

if I don't, the linker will refuse to link C++ dependencies (at least for aarch64). I wonder if there's a possibility to include this in the SDK settings somehow? Or if this needs to be built into swiftc more deeply?

Reply:
It appears this is a regression in this new Swift 6 SDK bundle, which @lhoward pointed out before. You can use that workaround on the command-line for now, and I will look into adding it to the bundle with the next release.